src/main/dolphinManager.ts

```diff
@@ -0,0 +1,216 @@
+import { ChildProcessWithoutNullStreams, spawn } from "child_process";
+import { randomBytes } from "crypto";
+import { app } from "electron";
+import electronSettings from "electron-settings";
+import { EventEmitter } from "events";
+import * as fs from "fs-extra";
+import { find, remove } from "lodash";
+import path from "path";
+import { DolphinType, findDolphinExecutable } from "common/dolphin";
+
+electronSettings.configure({
+  fileName: "Settings",
+  prettify: true,
+});
+
+export interface ReplayCommunication {
+  mode: "normal" | "mirror" | "queue"; // default normal
+  replay?: string; // path to the replay if in normal or mirror mode
+  startFrame?: number; // when to start watching the replay
+  endFrame?: number; // when to stop watching the replay
+  commandId?: string; // random string, doesn't really matter
+  outputOverlayFiles?: boolean; // outputs gameStartAt and gameStation to text files (only works in queue mode)
+  isRealTimeMode?: boolean; // default true; keeps dolphin fairly close to real time (about 2-3 frames); only relevant in mirror mode
+  shouldResync?: boolean; // default true; disables the resync functionality
+  rollbackDisplayMethod?: "off" | "normal" | "visible"; // default off; normal shows like a player experienced it, visible shows ALL frames (normal and rollback)
+  queue?: ReplayQueueItem[];
+}
+
+export interface ReplayQueueItem {
+  path: string;
+  startFrame?: number;
+  endFrame?: number;
+  gameStartAt?: string;
+  gameStation?: string;
+}
+
+interface DolphinInstance {
+  dolphinUseType: "playback" | "spectate" | "config" | "netplay";
+  index?: number; // should refer to the index in relation to the list of sources for spectating
+  dolphin?: ChildProcessWithoutNullStreams;
+  commFilePath?: string;
+}
+
+interface DolphinInstances {
+  playback: DolphinInstance | null;
+  spectate: DolphinInstance[] | null;
+  netplay: DolphinInstance | null;
+  configNetplay: DolphinInstance | null;
+  configPlayback: DolphinInstance | null;
+}
+
+async function generateCommunicationFilePath(
+  dolphinUseType: "playback" | "spectate" | "config" | "netplay",
+): Promise<string> {
+  const tmpDir = app.getPath("temp");
+  const uniqueId = randomBytes(3 * 4).toString("hex");
+  const commFileName = `slippi-${dolphinUseType}-${uniqueId}.txt`;
+  const commFileFullPath = path.join(tmpDir, commFileName);
+
+  return commFileFullPath;
+}
+
+// DolphinManager should be in control of all dolphin instances that get opened for actual use.
+// This includes playing netplay, viewing replays, watching broadcasts (spectating), and configuring Dolphin.
+// The openDolphin function is exported so other parts of the app can do some checks.
+export class DolphinManager extends EventEmitter {
```

Comment:
Is there a reason why we extend `EventEmitter` but we don't actually emit any events with `this.emit()`?